pthread_util.c

```diff
@@ -33,10 +33,13 @@ struct pthread_osx_trampoline_args {
 
 void *pthread_osx_trampoline(void *args) {
 	printf("Trampoline executed!\n");
-	struct pthread_osx_trampoline_args *t_args = (struct pthread_osx_trampoline_args *) args;
-	pthread_setname_np(t_args->thread_name);
+    struct pthread_osx_trampoline_args *t_args = (struct pthread_osx_trampoline_args *) args;
+	const char *thread_name = t_args->thread_name;
+    void *thread_args = t_args->thread_args;
+    pthread_setname_np(thread_name);
 	printf("Thread name: %s\n", t_args->thread_name);
-	t_args->thread_start(t_args->thread_args);
+    free(args);
+	t_args->thread_start(thread_args);
```

Comment:
Uh, t_args == args, so if you free(args), t_args is no longer valid.  The free needs to happen after t_args->thread_start(thread_args);

Reply:
Whoops. Typo on my part...

Reply:
Force pushed a fix to keep the commit log clean. Updated to fix.